Sources/PlaydateKit/Core/Sound.swift

```diff
@@ -44,6 +50,37 @@ public enum Sound {
         @discardableResult public func play(repeat: Int32 = 1) -> Int32 {
             fileplayer.play.unsafelyUnwrapped(pointer, `repeat`)
         }
+        
+        private struct CallbackData {
+            var callback: (() -> Void)?
+        }
+        private var _callbackData: UnsafeMutablePointer<CallbackData>?
+
+        /// Installs a closure that will be called when playback has completed. This is implemented
+        /// by wrapping the closure in a struct that is stored in a dynamically allocated block, which
+        /// is passed as `userData` to `setFinishCallback` below.  
+        public var finishCallback: (() -> Void)? {
+            get {
+                _callbackData?.pointee.callback
+            }
+            set {
+                if let callbackData = _callbackData {
+                    callbackData.pointee.callback = newValue
+                } else if let newValue {
+                    let callbackData: UnsafeMutablePointer<CallbackData> = .allocate(capacity: 1)
+                    callbackData.initialize(to: .init(callback: newValue))
+                    _callbackData = callbackData
+                    setFinishCallback(
+                        callback: { sourceSource, userdata in
+                            if let callback = userdata?.assumingMemoryBound(to: CallbackData.self).pointee.callback {
+                                callback()
+                            }
+                        },
+                        soundUserdata: callbackData
+                    )
+                }
+            }
+        }
```

Comment:
This seems like a nice approach for this. Alternatives would be to mark FilePlayer as open, set the userdata to a pointer to the file player, and have it call an open `didFinish()` method that people can override (similar to what I'm doing in #32). Another option might be to allow setting a delegate on the FilePlayer, which would align with iOS APIs.

Don't really have much of a preference here, just thinking out loud.

Reply:
Another approach might include the `FIlePlayer` as a parameter to the closure. This may be preferrable, to avoid creating cycles, since `weak` is unavailable in embedded Swift (which is a shame).

Reply:
One other tiny change might be to reorder the parameters of setFinishCallback() so a trailing closure can be used.